### PR TITLE
Test wasm32-unknown-unknown in Azure Pipelines

### PR DESCRIPTION
**Pull request by heaths**
Also gets rid of most of GitHub Actions with only E2E tests remaining.

### CODE REVIEW

**Comment by heaths** [comment]
@LarryOsterman can we even run E2E tests against Event Hubs? That requires actually-provisioned resources, right? If not, I can delete the entirety of the GitHub Actions workflows finally. /cc @hallipr

**Comment by heaths** [comment]
@microsoft-github-policy-service rerun

**Comment by LarryOsterman** [comment]
> @LarryOsterman can we even run E2E tests against Event Hubs? That requires actually-provisioned resources, right? If not, I can delete the entirety of the GitHub Actions workflows finally. /cc @hallipr

No, the test_e2e tests for Event Hubs requires either the eventhub emulator running locally or actually provisioned resources. It's a live test in the classic vernacular. 

As we get further into ADO pipelines, I'll probably bring up the test AMQP Broker (used in Go) to test the AMQP stuff locally, that will help with coverage.

**Review by LarryOsterman — APPROVED** [review]

File: eng/scripts/Test-Packages.ps1 [hunk 1]
```diff
@@ -30,22 +31,24 @@ if ($PackageInfoPath) {
   | ConvertFrom-Json
 }
 else {
-  $packagesToTest = Get-AllPackagesInRepo
+  . "$PSScriptRoot\Language-Settings.ps1"
```
Comment (hallipr) [inline eng/scripts/Test-Packages.ps1:34]: Line 14's invocation of eng/common/scripts/common.ps1 will import Language-Settings.ps1

File: eng/scripts/Test-Packages.ps1 [hunk 2]
```diff
@@ -30,22 +31,24 @@ if ($PackageInfoPath) {
...
+  Write-Host "  '$($package.Name)' in '$($package.DirectoryPath)'"
 }
 
 Write-Host "Setting RUSTFLAGS to '-Dwarnings'"
 $env:RUSTFLAGS = "-Dwarnings"
 
 
 foreach ($package in $packagesToTest) {
-  Push-Location (Join-Path $RepoRoot $package.DirectoryPath)
+  Push-Location ([System.IO.Path]::Combine($RepoRoot, $package.DirectoryPath))
```
Comment (hallipr) [inline eng/scripts/Test-Packages.ps1:49]: Were you seeing a difference between Path::Combine and Join-Path here?

Comment (hallipr) [reply eng/scripts/Test-Packages.ps1:49]: Nevermind.  I saw your note re: combining absolute paths:
```
> Join-Path  'c:\something\else' 'd:\another\path'
c:\something\else\d:\another\path

> [IO.Path]::Combine('c:\something\else', 'd:\another\path')
d:\another\path
```

Comment (heaths) [reply eng/scripts/Test-Packages.ps1:49]: Yeah. Never use `Join-Path`. It's utter garbage and gets people into trouble so many times. It's worse on non-Windows where you don't get a drive letter in the middle of two absolute paths - just `/`s.

**Pull request merged**